src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -99,6 +99,10 @@ export class KeyStrokeHandler {
                     break
                 }
             }
+            if (this.keyStrokeCount >= 15) {
+                triggerType = 'KeyStrokeCount'
+                this.keyStrokeCount = 0
+            }
```

Comment:
Should this logic be within the last case `DocumentChangedSource.RegularKey:`? Otherwise the triggerType will always be overridden to `KeyStrokeCount` when the count is 15 for any `DocumentChangedSource`. And this will impact telemetry and anything that uses this value.

Reply:
Makes sense.